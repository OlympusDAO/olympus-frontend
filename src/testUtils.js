import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "@material-ui/core/styles";
import CssBaseline from "@material-ui/core/CssBaseline";
import { light as lightTheme } from "./themes/light.js";
import { Web3ContextProvider } from "./hooks/web3Context";
import store from "./store.ts";
import { createMemoryHistory } from "history";
import App from "./App";
import { I18nProvider } from "@lingui/react";
import { i18n } from "@lingui/core";

const AllTheProviders = ({ children }) => {
  return (
    <Web3ContextProvider>
      <Provider store={store}>
        <BrowserRouter basename={"/"}>
          <ThemeProvider theme={lightTheme}>
            <CssBaseline />
            {children}
          </ThemeProvider>
        </BrowserRouter>
      </Provider>
    </Web3ContextProvider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

const renderRoute = function (route) {
  const history = createMemoryHistory();
  history.push(route);
  render(
    <Web3ContextProvider>
      <Provider store={store}>
        <I18nProvider i18n={i18n}>
          <BrowserRouter basename={"/#"}>
            <App />
          </BrowserRouter>
        </I18nProvider>
      </Provider>
    </Web3ContextProvider>,
  );
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderRoute };
