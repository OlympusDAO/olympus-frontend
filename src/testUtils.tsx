import { i18n } from "@lingui/core";
import { I18nProvider } from "@lingui/react";
import CssBaseline from "@material-ui/core/CssBaseline";
import { ThemeProvider } from "@material-ui/core/styles";
import { render, RenderOptions, RenderResult } from "@testing-library/react";
import { createMemoryHistory } from "history";
import React, { ReactElement, ReactNode } from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

import App from "./App";
import { Web3ContextProvider } from "./hooks/web3Context";
import { ReactQueryProvider } from "./lib/react-query";
import defaultStore from "./store";
import { light as lightTheme } from "./themes/light.js";

const customRender = (ui: ReactElement, store = defaultStore, options?: RenderOptions): RenderResult => {
  const ProviderWrapper = ({ children }: { children?: ReactNode }) => (
    <Web3ContextProvider>
      <ReactQueryProvider>
        <Provider store={store}>
          <I18nProvider i18n={i18n}>
            <BrowserRouter basename={"/"}>
              <ThemeProvider theme={lightTheme}>
                <CssBaseline />
                {children}
              </ThemeProvider>
            </BrowserRouter>
          </I18nProvider>
        </Provider>
      </ReactQueryProvider>
    </Web3ContextProvider>
  );
  return render(ui, { wrapper: ProviderWrapper, ...options });
};

const renderRoute = function (route: string, store = defaultStore) {
  const history = createMemoryHistory();
  history.push(route);
  return render(
    <Web3ContextProvider>
      <ReactQueryProvider>
        <Provider store={store}>
          <I18nProvider i18n={i18n}>
            <BrowserRouter basename={"/"}>
              <App />
            </BrowserRouter>
          </I18nProvider>
        </Provider>
      </ReactQueryProvider>
    </Web3ContextProvider>,
  );
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderRoute };
