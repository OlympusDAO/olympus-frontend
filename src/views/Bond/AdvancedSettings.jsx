import {
  Typography,
  Box,
  Modal,
  Paper,
  SvgIcon,
  IconButton,
  FormControl,
  OutlinedInput,
  InputLabel,
  InputAdornment,
} from "@material-ui/core";
import { ReactComponent as XIcon } from "../../assets/icons/x.svg";
import "./bondSettings.scss";

function AdvancedSettings({
  open,
  handleClose,
  slippage,
  recipientAddress,
  onRecipientAddressChange,
  onSlippageChange,
}) {
  return (
    <Modal id="hades" open={open} onClose={handleClose} hideBackdrop>
      <Paper className="ohm-card ohm-popover">
        <Box display="flex">
          <IconButton onClick={handleClose}>
            <SvgIcon color="primary" component={XIcon} />
          </IconButton>
          <Typography variant="h3">Hades</Typography>
        </Box>

        <Box className="card-content">
          <InputLabel htmlFor="slippage">Slippage</InputLabel>
          <FormControl variant="outlined" color="primary" fullWidth>
            <OutlinedInput
              id="slippage"
              value={slippage}
              onChange={onSlippageChange}
              type="number"
              max="100"
              min="100"
              endAdornment={<InputAdornment position="end">%</InputAdornment>}
            />
            <div className="help-text">
              <Typography variant="body2" color="textSecondary">
                Transaction may revert if price changes by more than slippage %
              </Typography>
            </div>
          </FormControl>

          <InputLabel htmlFor="recipient">Recipient Address</InputLabel>
          <FormControl variant="outlined" color="primary" fullWidth>
            <OutlinedInput id="recipient" value={recipientAddress} onChange={onRecipientAddressChange} type="text" />
            <div className="help-text">
              <Typography variant="body2" color="textSecondary">
                Choose recipient address. By default, this is your currently connected address
              </Typography>
            </div>
          </FormControl>
        </Box>
      </Paper>
    </Modal>
  );
}

export default AdvancedSettings;
