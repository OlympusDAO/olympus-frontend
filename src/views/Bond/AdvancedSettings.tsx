import "./bondSettings.scss";

import { Trans } from "@lingui/macro";
import {
  Box,
  FormControl,
  IconButton,
  InputAdornment,
  InputLabel,
  Modal,
  OutlinedInput,
  Paper,
  SvgIcon,
  Typography,
} from "@material-ui/core";
import { ChangeEvent } from "react";

import { ReactComponent as XIcon } from "../../assets/icons/x.svg";

interface IAdvancedSettingsProps {
  readonly open: boolean;
  readonly recipientAddress: string;
  readonly slippage: string;
  readonly handleClose: () => void;
  readonly onRecipientAddressChange: (e: ChangeEvent<HTMLInputElement>) => void;
  readonly onSlippageChange: (e: ChangeEvent<HTMLInputElement>) => void;
}

function AdvancedSettings(props: IAdvancedSettingsProps) {
  return (
    <Modal id="hades" open={props.open} onClose={props.handleClose} hideBackdrop>
      <Paper className="ohm-card ohm-popover">
        <Box display="flex">
          <IconButton onClick={props.handleClose}>
            <SvgIcon color="primary" component={XIcon} />
          </IconButton>
          <Typography variant="h3">Settings</Typography>
        </Box>
        <Box className="card-content">
          <InputLabel htmlFor="slippage">
            <Trans>Slippage</Trans>
          </InputLabel>
          <FormControl variant="outlined" color="primary" fullWidth>
            <OutlinedInput
              id="slippage"
              value={props.slippage}
              onChange={props.onSlippageChange}
              type="number"
              endAdornment={<InputAdornment position="end">%</InputAdornment>}
            />
            <div className="help-text">
              <Typography variant="body2" color="textSecondary">
                <Trans>Transaction may revert if price changes by more than slippage %</Trans>
              </Typography>
            </div>
          </FormControl>

          <InputLabel htmlFor="recipient">
            <Trans>Recipient Address</Trans>
          </InputLabel>
          <FormControl variant="outlined" color="primary" fullWidth>
            <OutlinedInput
              id="recipient"
              value={props.recipientAddress}
              onChange={props.onRecipientAddressChange}
              type="text"
            />
            <div className="help-text">
              <Typography variant="body2" color="textSecondary">
                <Trans>Choose recipient address. By default, this is your currently connected address</Trans>
              </Typography>
            </div>
          </FormControl>
        </Box>
      </Paper>
    </Modal>
  );
}

export default AdvancedSettings;
