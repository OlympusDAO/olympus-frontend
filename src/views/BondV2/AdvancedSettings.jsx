import "./bondSettings.scss";

import { Trans } from "@lingui/macro";
import { Box, FormControl, InputAdornment, InputLabel, OutlinedInput, Typography } from "@material-ui/core";
import { Modal } from "@olympusdao/component-library";

function AdvancedSettings({
  open,
  handleClose,
  slippage,
  recipientAddress,
  onRecipientAddressChange,
  onSlippageChange,
}) {
  return (
    <Modal
      id="hades"
      open={open}
      headerText={"Settings"}
      minHeight={"400px"}
      closePosition={"left"}
      onClose={handleClose}
      maxWidth={"500px"}
    >
      <Box className="card-content">
        <InputLabel htmlFor="slippage">
          <Trans>Slippage</Trans>
        </InputLabel>
        <FormControl variant="outlined" color="primary" fullWidth>
          <OutlinedInput
            id="slippage"
            value={slippage}
            onChange={onSlippageChange}
            type="number"
            max="100"
            min="100"
            endAdornment={<InputAdornment position="end">%</InputAdornment>}
          />
          <Typography variant="body2" color="textSecondary" className={"helper-text"}>
            <Trans>Transaction may revert if price changes by more than slippage %</Trans>
          </Typography>
        </FormControl>

        <InputLabel htmlFor="recipient">
          <Trans>Recipient Address</Trans>
        </InputLabel>
        <FormControl variant="outlined" color="primary" fullWidth>
          <OutlinedInput id="recipient" value={recipientAddress} onChange={onRecipientAddressChange} type="text" />
          <Typography variant="body2" color="textSecondary" className={"helper-text"}>
            <Trans>Choose recipient address. By default, this is your currently connected address</Trans>
          </Typography>
        </FormControl>
      </Box>
    </Modal>
  );
}

export default AdvancedSettings;
