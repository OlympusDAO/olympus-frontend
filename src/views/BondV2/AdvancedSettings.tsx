import "./BondSettings.scss";

import { Trans } from "@lingui/macro";
import { Box, FormControl, InputAdornment, InputLabel, OutlinedInput, Typography } from "@material-ui/core";
import { Modal } from "@olympusdao/component-library";
import { ChangeEvent } from "react";

interface IAdvancedSettingsProps {
  readonly open: boolean;
  readonly recipientAddress: string;
  readonly slippage: number;
  readonly handleClose: () => void;
  readonly onRecipientAddressChange: (e: ChangeEvent<HTMLInputElement>) => void;
  readonly onSlippageChange: (e: ChangeEvent<HTMLInputElement>) => void;
}

function AdvancedSettings(props: IAdvancedSettingsProps) {
  return (
    <Modal
      id="hades"
      open={props.open}
      onClose={props.handleClose}
      maxWidth="300"
      minHeight="350"
      headerText={"Settings"}
    >
      <Box className="card-content">
        <InputLabel htmlFor="slippage">
          <Trans>Slippage</Trans>
        </InputLabel>
        <FormControl variant="outlined" color="primary" fullWidth>
          <OutlinedInput
            id="slippage"
            value={props.slippage}
            onChange={props.onSlippageChange}
            type="number"
            endAdornment={<InputAdornment position="end">%</InputAdornment>}
          />
          <div className="helper-text">
            <Typography variant="body2" color="textSecondary">
              <Trans>Transaction may revert if price changes by more than slippage %</Trans>
            </Typography>
          </div>
        </FormControl>

        <InputLabel htmlFor="recipient">
          <Trans>Recipient Address</Trans>
        </InputLabel>
        <FormControl variant="outlined" color="primary" fullWidth>
          <OutlinedInput
            id="recipient"
            value={props.recipientAddress}
            onChange={props.onRecipientAddressChange}
            type="text"
          />
          <div className="helper-text">
            <Typography variant="body2" color="textSecondary">
              <Trans>Choose recipient address. By default, this is your currently connected address</Trans>
            </Typography>
          </div>
        </FormControl>
      </Box>
    </Modal>
  );
}

export default AdvancedSettings;
