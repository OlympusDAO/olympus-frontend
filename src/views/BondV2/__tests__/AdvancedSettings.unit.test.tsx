import { render } from "../../../testUtils";
import AdvancedSettings from "../AdvancedSettings";

describe("<AdvancedSettings/>", () => {
  it("should render component", () => {
    const { container } = render(
      <AdvancedSettings
        open={true}
        recipientAddress={"123"}
        slippage={0.05}
        handleClose={() => console.log("close")}
        onRecipientAddressChange={() => console.log("onRecipientAddressChange")}
        onSlippageChange={() => console.log("onSlippageChange")}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
