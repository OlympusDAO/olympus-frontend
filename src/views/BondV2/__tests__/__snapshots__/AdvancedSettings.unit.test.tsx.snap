// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdvancedSettings/> should render component 1`] = `
<div
  aria-hidden="true"
/>
`;
